### PR TITLE
fix and update binary cross entropy loss tests

### PR DESCRIPTION
**Pull request by m0saan**
I inadvertently failed to push the latest version for the binary cross-entropy loss tests. @awni, could you kindly perform a quick review and approve? Apologies for any redundancy.

### CODE REVIEW

**Comment by awni** [comment]
I think you need to rebase as I pushed a fix for part of this already and there is a merge conflict. Also make sure you run the pre-commit formatting if you did not.

**Comment by m0saan** [comment]
@awni I did a rebase now!

File: python/tests/test_nn.py [hunk 1]
```diff
@@ -101,23 +101,75 @@ def test_kl_div_loss(self):
...
+                ],
+            ]
+        )
+        self.assertTrue(mx.allclose(losses_none, expected_none, rtol=1e-5, atol=1e-8))
+
+        # Test with reduction 'mean'
+        loss_module_mean = nn.losses.BCELoss(reduction="mean")
+        losses_mean = loss_module_mean(inputs, targets)
+        expected_mean = mx.mean(expected_none)
+        self.assertEqual(losses_mean.item(), expected_mean.item())
```
Comment (awni) [inline python/tests/test_nn.py:166]: I think these should be `self.assertTrue(mx.allclose(losses_mean, expected_mean))` otherwise they fail in some cases due to subtle numerical differences.

File: python/tests/test_nn.py [hunk 2]
```diff
@@ -101,23 +101,75 @@ def test_kl_div_loss(self):
...
+
+        # Test with reduction 'sum'
+        loss_module_sum = nn.losses.BCELoss(reduction="sum")
+        losses_sum = loss_module_sum(inputs, targets)
+        expected_sum = mx.sum(expected_none)
+        self.assertEqual(losses_sum.item(), expected_sum.item())
```
Comment (awni) [inline python/tests/test_nn.py:172]: Same here.

File: python/tests/test_nn.py [hunk 3]
```diff
@@ -101,23 +101,75 @@ def test_kl_div_loss(self):
...
+                ],
+            ]
+        )
+        self.assertTrue(mx.allclose(losses_none, expected_none, rtol=1e-5, atol=1e-8))
 
         # Test with reduction 'mean'
         losses_mean = nn.losses.binary_cross_entropy(inputs, targets, reduction="mean")
         expected_mean = mx.mean(expected_none)
-        self.assertTrue(mx.allclose(losses_mean, expected_mean))
+        self.assertEqual(losses_mean.item(), expected_mean.item())
```
Comment (awni) [inline python/tests/test_nn.py:130]: Same here

File: python/tests/test_nn.py [hunk 4]
```diff
@@ -101,23 +101,75 @@ def test_kl_div_loss(self):
...
 
         # Test with reduction 'sum'
         losses_sum = nn.losses.binary_cross_entropy(inputs, targets, reduction="sum")
         expected_sum = mx.sum(expected_none)
-        self.assertTrue(mx.allclose(losses_sum, expected_sum))
+        self.assertEqual(losses_sum.item(), expected_sum.item())
```
Comment (awni) [inline python/tests/test_nn.py:135]: Same here.

**Review by awni — COMMENTED** [review]
Minor comments then we can merge. In general prefer allclose for floating point comparisons or else numerical differences can cause the tests to fail when they are run in different environments / devices, etc.

File: python/tests/test_nn.py (hunk 1 above)
Comment (m0saan) [reply python/tests/test_nn.py]: done!

File: python/tests/test_nn.py (hunk 2 above)
Comment (m0saan) [reply python/tests/test_nn.py]: done!

**Review by awni — APPROVED** [review]
Thanks a ton!
